go/minimal/main.go

```diff
@@ -96,12 +93,19 @@ func main() {
 		panic("failed to create subscribe stream: " + err.Error())
 	}
 
-	for msg := range events {
-		var m MessageInABottle
-		if err := json.Unmarshal(msg.Data, &m); err != nil {
-			panic(fmt.Errorf("failed to unmarshal message: %s", err))
+	go func() {
+		for msg := range events {
+			var m MessageInABottle
+			if err := json.Unmarshal(msg.Data, &m); err != nil {
+				panic(fmt.Errorf("failed to unmarshal message: %s", err))
+			}
+			fmt.Printf("At %s,\n%s\nsent you the following message...\n'%s'\n", m.Timestamp, m.Sender, m.Message)
+			return
 		}
-		fmt.Printf("At %s,\n%s\nsent you the following message...\n'%s'\n", m.Timestamp, m.Sender, m.Message)
-		return
-	}
+	}()
+
+	// Publish event
+	time.Sleep(1 * time.Second)
+	pub.Publish(topicID, e)
+	time.Sleep(1 * time.Second)
```

Comment:
Putting this in a go routine was my attempt to see if the issue was out of order messaging.